symengine/sympy_compat.py

```diff
@@ -43,6 +67,137 @@ def __new__(cls, i):
         return symengine.Integer(i)
 
 
+class Float(Number):
+    _classes = (symengine.RealDouble, symengine.RealMPFR)
+
+    def __new__(cls, num, dps=None, prec=None, precision=None):
+
+        if dps is not None and precision is not None:
+            raise ValueError('Both decimal and binary precision supplied. '
+                             'Supply only one. ')
+
+        if isinstance(num, (str, unicode)):
+            num = num.replace(' ', '')
+            if num.startswith('.') and len(num) > 1:
+                num = '0' + num
+            elif num.startswith('-.') and len(num) > 2:
+                num = '-0.' + num[2:]
+        elif isinstance(num, float) and num == 0:
+            num = '0'
+        elif isinstance(num, (int, long, Integer)):
+            num = str(num)
+        elif isinstance(num, (sympy.Infinity, sympy.NegativeInfinity)):
+            raise NotImplementedError
+        '''
+        elif isinstance(num, mpmath.mpf):
+            if precision is None:
+                if dps is None:
+                    precision = num.context.prec
+            num = num._mpf_
+        '''
+        
+        if dps is None and precision is None:
+            dps = 15
+            if isinstance(num, Float):
+                return num
+            if isinstance(num, str) and _literal_float(num):
+                try:
+                    Num = decimal.Decimal(num)
+                except decimal.InvalidOperation:
+                    pass
+                else:
+                    isint = '.' not in num
+                    num, dps = _decimal_to_Rational_prec(Num)
+                    if num.is_Integer and isint:
+                        dps = max(dps, len(str(num).lstrip('-')))    
+                    dps = max(15, dps)
+                    '''
+                    precision = mlib.libmpf.dps_to_prec(dps)
```

Comment:
@isuruf Would you be fine with re-implementing `dps_to_prec` in `symengine.py`? Or would it cause a problem due to the licenses?